libyul/YulStack.cpp

```diff
@@ -164,14 +166,39 @@ void YulStack::optimize(Object& _object, bool _isCreation)
 	if (EVMDialect const* evmDialect = dynamic_cast<EVMDialect const*>(&dialect))
 		meter = std::make_unique<GasMeter>(*evmDialect, _isCreation, m_optimiserSettings.expectedExecutionsPerDeployment);
 
+	auto [optimizeStackAllocation, yulOptimiserSteps, yulOptimiserCleanupSteps] = [&]() -> std::tuple<bool, std::string, std::string>
+	{
+		if (!m_optimiserSettings.runYulOptimiser)
+		{
+			// Yul optimizer disabled, but empty sequence (:) explicitly provided
+			if (OptimiserSuite::isEmptyOptimizerSequence(m_optimiserSettings.yulOptimiserSteps + ":" + m_optimiserSettings.yulOptimiserCleanupSteps))
```

Comment:
But wait, shouldn't we also have some sort of alias from `--no-optimize-yul` to `--yul-optimizations=":"`? I mean, if I got this correctly, `solc --yul --no-optimize-yul x.yul` still being optimized but `solc --yul --no-optimize-yul --yul-optimizations=":" x.yul` is now allowed?

Reply:
The way it is implemented currently, `--no-optimize-yul` is an equivalent of `yul: false`. And that one does not automatically set an empty sequence either.

This may change when we decide how to rework the optimizer options, but on the call today we decided to go with the PR as defined orignally and in that context this change does not make sense.

BTW, to compile Yul use `--strict-assembly`. `--yul` is for the typed Yul dialect, which is now obsolete :)